hashtagsv2/hashtags/static/js/graph.js

```diff
@@ -118,47 +118,109 @@ else{
     })
 }
 
+// calling the edits_over_time API when page loads gives default view_type
 $.ajax({
     url: '/api/time_stats/?' + url_param,
     dataType: 'json',
-    success: function(data){
-        var ctx = document.getElementById('timeChart');
-        var timeChart = new Chart(ctx, {
-            type: 'line',
-            data: {
-                labels: data['time_array'],
-                datasets: [{
-                    label: 'Number of edits',
-                    data: data['edits_array'],
-                    backgroundColor: 'rgba(255, 99, 132, 0.2)',    
-                    borderColor: 'rgba(255, 99, 132, 1)',
-                    borderWidth: 1,
-                    fill: false
-                }]
+    success: function(data) {
+        show_chart(data['view_type']);
+        draw_chart(data);
+    },
+    complete: function() {
+        $('#loader3').hide();
+    }
+})
+
+
+// when user clicks on button to change default view_type
+$(".view-type").click(function() {
+    var chart_name = $(this).attr("name");
+    var status = $('#' + chart_name).attr("status");
+
+    // if chart is loaded previously
+    if (status == "loaded") {
+        show_chart(chart_name);
+    } else {
+        var view_type = $(this).attr("value");
+        //adding view_type parameter to url
+        extra_url_param = url_param + '&amp;view_type=' + view_type
+
+        $.ajax({
+            url: '/api/time_stats/?' + extra_url_param,
+            dataType: 'json',
+            success: function(data) {
+                show_chart(data['view_type']);
+                draw_chart(data);
             },
-            options: {
-                scales: {
-                    yAxes: [{
-                        ticks: {
-                            beginAtZero: true,
-                            callback: function (value) { if (Number.isInteger(value)) { return value; } }
+            complete: function() {
+                $('#loader3').hide();
+            }
+        })
+    }
+});
+
+// shows the required chart and hides oher
+function show_chart(view_type) {
+    $(".timeChart").hide();
+    $('#' + view_type).show();
+
+    for (var i = 0; i < 3; i++) {
+        ($(".view-type")[i]).style.borderStyle = "hidden";
+    }
+
+    document.getElementsByName(view_type)[0].style.borderStyle = "solid";
+
+    var status = $('#' + view_type).attr("status");
```

Comment:
I don't think we need the logic from line 173 to 180 because we are performing the same in the `draw_chart` function after the animation is completed.

Reply:
Therefore, we can also remove the `status` attribute from the canvas element in graph.html.

Reply:
The `status` attribute is added so the web page doesn't have to call an API every single time the user changes the view-type in same session. The `show_chart` is used to hide all chart (now download JPEG buttons also) and display only the required one by the user and `draw_chart` draws a chart from scratch, from data received through an API call.
When the page loads the default view chart get drawn and displayed. And  when user clicks other options it calls an API and draws and displays that chart. 
But the catch here is that when it draws for the first time its `status` attribute is set to _loaded_. So next time when user clicks on that option it doesn't have to call an API again, but simple hide all the other charts and displays the required one. 
I added this functionality, as I observed that for the charts with number of days over 300-400, daily- view data can take significant time to load (observation form live project). And loading all data (three view-types) together also doesn't seems to be a smart option.

Regarding the logic from line 173-180. Earlier there was only one Download JPEG button and it needs to be changed regularly with the change in view-type by user. Therefore line 173-180 are used again. Now in the last 2 commits, I added three different buttons for different views, they will be displayed and hidden same as the respective charts.These commits will make the code more readable in my understanding.

Reply:
Ohh yes, keeping `status` attribute would be fine. I just meant to remove the repeated logic behind the JPEG download, but now you have removed it in the latest commit. So it is fine. :)

Reply:
Yep, I earlier there was only one button, so the logic needs to be repeated every time view is changed . But now I have added three buttons, so need of repetition. :)

Reply:
Great work! maybe I misunderstood it before. But now it looks good.